lib/bolt/transport/ssh.rb

```diff
@@ -115,8 +115,12 @@ def run_script(target, script, arguments, options = {})
       end
 
       def run_task(target, task, arguments, options = {})
-        executable = target.select_impl(task, PROVIDED_FEATURES)
-        raise "No suitable implementation of #{task.name} for #{target.name}" unless executable
+        unless task.respond_to? :file_content
```

Comment:
This section is kind of hard to reason about. I'd also like to unify these checks and the section below.

I'd lean towards
```
if task.respond_to?(:file_content) and task.file_content
  filename = task.name.split("::").last
  write_executable = lambda { |conn, dir| conn.write_executable_from_content(dir, task.file_content, filename) }
else
  executable = target.select_impl(task, PROVIDED_FEATURES)
  raise "No suitable implementation of #{task.name} for #{target.name}" unless executable
  write_executable = lambda { |conn, dir| conn.write_remote_executable(dir, executable) }
end
...
remote_task_path = write_executable(conn, dir)
```